### PR TITLE
Refactor changes from core

### PR DESCRIPTION
**Pull request by DarshitChanpura**
### Description
Pull latest refactor changes from core

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/sdk/SDKClient.java [hunk 1]
```diff
@@ -60,21 +53,7 @@
...
-import org.opensearch.client.Node;
-import org.opensearch.client.RequestOptions;
-import org.opensearch.client.Requests;
-import org.opensearch.client.RestClient;
-import org.opensearch.client.Request;
-import org.opensearch.client.Response;
-import org.opensearch.client.ResponseListener;
-import org.opensearch.client.RestClientBuilder;
-import org.opensearch.client.RestHighLevelClient;
+import org.opensearch.client.*;
```
Comment (owaiskazi19) [inline src/main/java/org/opensearch/sdk/SDKClient.java:56]: Can you take care of the wildcard import?

Comment (dbwiddis) [reply src/main/java/org/opensearch/sdk/SDKClient.java:56]: But I like stars! :) 

image

Comment (DarshitChanpura) [reply src/main/java/org/opensearch/sdk/SDKClient.java:56]: I swear everytime I forget to run `./gradlew spotlessApply` intelli J sneaks in this `*` imports

**Review by owaiskazi19 — APPROVED** [review]

**Comment by dbwiddis** [comment]
Gah! I'm sorry @DarshitChanpura i just updated some of these imports, forgetting you had this PR open.

Looks like you covered more than I did, can you rebase this?

**Comment by dbwiddis** [comment]
Also if you want to throw in standard import ordering to align with recent OpenSearch changes I wouldn't mind.

**Comment by owaiskazi19** [comment]
> Also if you want to throw in standard import ordering to align with recent OpenSearch changes I wouldn't mind.

Something like the below for spotless:

```
importOrder(
              'de.thetaphi',
              'com.carrotsearch',
              'com.fasterxml',
              'com.avast',
              'com.sun',
              'com.maxmind|com.github|com.networknt|groovy|nebula',
              'org.antlr',
              'software.amazon',
              'com.azure|com.microsoft|com.ibm|com.google|joptsimple|org.apache|org.bouncycastle|org.codehaus|org.opensearch|org.objectweb|org.joda|org.hamcrest|org.openjdk|org.gradle|org.junit',
              'javax',
              'java',
              '',
              '\\#java|\\#org.opensearch|\\#org.hamcrest|\\#'
          )

```

as mentioned here: https://github.com/opensearch-project/OpenSearch/issues/9306

**Review by owaiskazi19 — APPROVED** [review]
:ship:

**Review by dbwiddis — APPROVED** [review]

**Pull request merged**